### DISCUSSION TITLE
Taster zum Einschalten des Display (ansonsten aus) + Smart Display

### DISCUSSION THREAD

**Discussion by themanfrommoon** [tbnobody/OpenDTU#705] [Ideas]
Moin,

ich habe hier einen Verbesserungsvorschlag für die Displays:

Auf Knopfdruck wird das Display für z.B. 30 Sekunden (einstellbar) eingeschaltet.
Danach geht es wieder automatisch aus.
Das verhindert das einbrennen.
Es verlängert die Lebensdauer.
Es nervt nicht, weil es dauernd hell ist, z.B. in der Nacht.
Und es spart Strom.
Der Taster sollte optional sein.

Ob das Display dabei stromlos geschaltet wird, oder nur alle Pixel auf dunkel geschaltet werden ist dabei egal. Letzteres wird wahrscheinlich einfacher sein.

Beste Grüße,
Chris

**Comment by uvcollector** [comment]
Genau das brauche ich auch 🎉

**Comment by ghost** [comment]
Lt. #575 hat man sich dagegen entschieden, das wird es also voraussichtlich nicht geben.

Ich habe dort daraufhin vorgeschlagen, ersatzweise einen Schalter in die Vcc-Leitung des Displays zu legen, um das Display nach Belieben ein-/ausschalten zu können - ganz unabhängig von speziellen Vorkehrungen in der Software. Zumindest bei den SSD1306-OLEDs scheint das einwandfrei zu funktionieren, andere habe ich nicht getestet.

**Reply by themanfrommoon** [reply]
Da sollte man nochmal neu drüber nachdenken!

**Comment by themanfrommoon** [comment]
Ich habe noch eine Erweiterungsidee:

Smart Display Option: (an/aus)
Das Display ist dauer-an, wenn der Wechselrichter an ist.
Wenn der Wechselrichter aus ist, dann kann man das Display über einen Taster starten.

In Summe hat man dann folgende Optionen:
1. Display immer aus, nur auf Knopfdruck für eine einstellbare Zeit an. (Display only on button: an, Smart Display Option: aus)
2. Display immer an, wenn Wechselrichter an. Wenn Wechselrichter aus, dann nur auf Knopfdruck für eine einstellbare Zeit an. (Display only on button: (automatisch) an, Smart Display Option: an)
3. Display immer an. (Display only on button: aus, Smart Display Option: aus)

Ich denke das ist ein deutlicher Mehrwert.

Beste Grüße,
Chris
